packages/orbit-components/src/Text/helpers/getMargin.ts

```diff
@@ -0,0 +1,34 @@
+import type { Props } from "../types";
+
+const getMargin = (margin: Props["margin"]): { vars: object; classes: string[] } => {
+  if (typeof margin === "string")
+    return { vars: { "--text-margin": margin }, classes: ["m-[var(--text-margin)]"] };
+
+  if (typeof margin === "number")
+    return { vars: { "--text-margin": `${margin}px` }, classes: ["m-var(--text-margin)]"] };
+
+  const { top, bottom, left, right } = margin || {};
+  const cssVar = {};
+  const marginClasses: string[] = ["m-0"];
+
+  if (top) {
```

Comment:
could it be something like? 🤔 : 

```
const result = [top, bottom, left, right].filter(Boolean).reduce(pos => {
 ...
}, cssVar: {}, classes: []})
```

Reply:
I have considered that, but it can't be like that. If one of those properties is `0`, it will be filtered out. Besides that, I do find the code more readable with the explicit conditions and results applied.

Reply:
It's just an example, you can easily filter only undefined values and let falsy values like 0 be there.

Reply:
Right, but won't that require me to dynamically construct the class names? I am not seeing a more efficient and readable approach, honestly, but I am open to suggestions